pkg/api/annotations.go

```diff
@@ -162,6 +162,7 @@ func UpdateAnnotation(c *models.ReqContext, cmd dtos.UpdateAnnotationsCmd) Respo
 		EpochEnd: cmd.TimeEnd,
 		Text:     cmd.Text,
 		Tags:     cmd.Tags,
+		Data:     cmd.Data,
```

Comment:
A bit unsure whether this could be a breaking change.

Reply:
It could be. The docs say that it "Updates all properties". But it could still break things for some users. What do you suggest?

Reply:
@marefr @jeet-parekh I think according to the semantics of the REST PUT verb, the `Data` field should be updated according to what the client passes here. I guess the problem is that the current API doesn't prescribe a `data` field to be passed in the JSON payload, so if we update the API, a client that follows the old spec and doesn't pass `data` will overwrite the annotation data with `nil`.

Might this be a case where we want to increment the REST API version to avoid breaking clients?

Another alternative might be to require that the client passes a non-null value for this field in the JSON payload, to update the data value (i.e., to erase the data, pass `{}` instead of `null` in the payload). Then, old clients not passing the `data` field will be safe.

Reply:
That alternative approach sounds good. Should I wait for a second approval before changing the implementation?

Reply:
@marefr What do you think of my idea that the client has to pass an empty JSON object to delete this field, instead of `null` so we retain backwards compatibility?

Reply:
>a client that follows the old spec and doesn't pass data will overwrite the annotation data with nil.

Data is `nullable: false` so technically could never be overwritten unless it's non-null. So I think it makes sense to have the requirement to expect `{}` to clear the annotation data. Then again, would need to check if this is even a valid state for an annotation to still exist.